static/src/javascripts/projects/commercial/commercial-metrics.ts

```diff
@@ -0,0 +1,84 @@
+import { EventTimer } from '@guardian/commercial-core';
+import { log } from '@guardian/libs';
+import config_ from '../../lib/config';
+
+const endpoint =
+	'//performance-events.code.dev-guardianapis.com/commercial-metrics';
+
+// This is really a hacky workaround ⚠️
+const config = config_ as {
+	get: (s: string, d?: unknown) => unknown;
+};
+
+type CommercialMetrics = {
+	browser_id?: string;
+	page_view_id: string;
+	received_timestamp: string;
+	received_date: string;
+	platform: string;
+	metrics: Metrics[];
+	properties: Properties[];
+};
+
+type Metrics = {
+	name: string;
+	value: number;
+};
+
+type Properties = {
+	name: string;
+	value: string;
+};
+
+let logged = false;
+
+const isDev = Boolean(config.get('page.isDev', false));
+const devProperties: Properties[] = isDev
+	? [{ name: 'isDev', value: window.location.hostname }]
+	: [];
+
+const logData = (): void => {
+	if (logged) return;
+	if (!window.guardian.ophan) return;
```

Comment:
Is this line still needed here or should commercial metric functionality be independent of ophan?

Reply:
Oh ignore this, if `!window.guardian.ophan` we can't get the `page_view_id` 👍